cpp/TypedIndex.h

```diff
@@ -360,7 +360,7 @@ class TypedIndex : public Index {
 
     int start = 0;
     if (!ep_added) {
-      size_t id = ids.size() ? ids.at(0) : (currentLabel);
+      size_t id = ids.size() ? ids.at(0) : (currentLabel.fetch_add(1));
```

Comment:
Is `.fetch_add(1)` actually what we want throughout this change?  Wouldn't that result in us incrementing `currentLabel` multiple times where we actually only wanted to increment it once?

Reply:
Correct, I'm reverting this

Reply:
Actually not, my original implementation is correct. Basically here we assign an id, then here: